core-chart/templates/postcreatehooks/its-hub-init.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if .Values.InstallPCHs }}
+apiVersion: tenancy.kflex.kubestellar.org/v1alpha1
+kind: PostCreateHook
+metadata:
+  name: its-hub-init
+  labels:
+    kflex.kubestellar.io/cptype: its
+spec:
+  templates:
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    spec:
+      template:
+        spec:
+          initContainers:
+          - name: wait-for-config-incluster
+            image: quay.io/kubestellar/kubectl:{{.Values.KUBECTL_VERSION}}
+            command:
+              - sh
+              - -c
+              - |
+                echo "Waiting for config-incluster in Secret {{.ITSSecretName}}..."
+                until kubectl get secret "{{.ITSSecretName}}" -n {{.ControlPlaneName}}-system -o jsonpath="{.data.config-incluster}" | grep -q .; do
```

Comment:
This `kubectl` command runs with `$KUBECONFIG` referring to the thing it is intended to wait for: the in-cluster kubeconfig data member of the Secret holding kubeconfigs for accessing the ITS. This is bad because it is circular, worse because it is looking in the wrong place. The Secret holding the kubeconfigs for accessing the ITS is in the KubeFlex hosting cluster.